File: internal/models/device.go
package models

import (
	"github.com/google/uuid"
	"github.com/juevigrace/diva-server/storage"
)

type Device struct {
	ID        uuid.UUID
	Name      string
	CreatedAt int64
	UpdatedAt int64
}

func DeviceFromDB(row *storage.DivaDevice) *Device {
	return &Device{
		ID:        row.ID,
		Name:      row.Name,
		CreatedAt: row.CreatedAt,
		UpdatedAt: row.UpdatedAt,
	}
}

File: client/packages/shared/app-lib/diva-lib-models-core/src/commonMain/kotlin/io/github/juevigrace/diva/lib/models/device/Device.kt
@file:OptIn(ExperimentalJsExport::class)
@file:DivaJsExport

package io.github.juevigrace.diva.lib.models.device

import io.github.juevigrace.diva.core.DivaJsExport
import io.github.juevigrace.diva.lib.models.api.device.DeviceResponse
import kotlin.js.ExperimentalJsExport

data class Device(
    val id: String,
    val name: String,
    val createdAt: Long,
    val updatedAt: Long,
) {
    companion object {
        fun fromResponse(response: DeviceResponse): Device {
            return Device(
                id = response.id,
                name = response.name,
                createdAt = response.createdAt,
                updatedAt = response.updatedAt,
            )
        }
    }
}
